radixdlt/src/integration/java/com/radixdlt/consensus/BFTCheck.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.consensus;
+
+import io.reactivex.rxjava3.core.Observable;
+
+/**
+ * A running BFT check given access to network
+ */
+public interface BFTCheck {
+
+	/**
+	 * Creates an observable which runs assertions against a bft network
+	 * @param network network to check
+	 * @return observable to subscribe to enable checking
+	 */
+	Observable<Object> check(BFTTestNetwork network);
```

Comment:
Just going by the documentation here it's not 100% clear to me what is supposed to happen if an assertion fails. Throw an assertion exception? Propagate an error down the observable?